store/oracles.js

```diff
@@ -1,25 +1,31 @@
-import Vue from 'vue'
+import { loadNextPage } from './utils'
 
 export const state = () => ({
-  oracles: {}
+  oracles: [],
+  nextPageUrl: ''
 })
 
 export const mutations = {
   setOracles (state, oracles) {
-    for (let oracle of oracles) {
-      Vue.set(state.oracles, oracle.oracle, oracle)
-    }
+    state.oracles = [...state.oracles, ...oracles.data]
+    state.nextPageUrl = oracles.next
   }
 }
 
 export const actions = {
-  getOracles: async function ({ rootGetters: { middleware }, commit }, { page, limit }) {
+  getLatestOracles: async function ({ rootGetters: { middleware }, commit }) {
     try {
-      const oracles = await middleware.getActiveOracles({ page, limit })
-      commit('setOracles', oracles.data)
+      if (state.nextPageUrl) return
+      const oracles = await middleware.getActiveOracles()
+      commit('setOracles', oracles)
     } catch (e) {
       console.log(e)
       commit('catchError', 'Error', { root: true })
     }
+  },
+  getMoreOracles: async function ({ state: { nextPageUrl }, commit }) {
+    if (!nextPageUrl) return
+    const oracles = await loadNextPage(nextPageUrl)
+    commit('setOracles', oracles)
```

Comment:
Future step would be to extract a module that will manage `nextPageUrl`, `getLatest<>`, `getMore<>` for each entity. This would allow getting rid of a lot of duplicate code.